### PR TITLE
This commit adds the POST /api/articles/:article_id/comments endpoint…

### PR DESCRIPTION
**Pull request by mjj677**
… and the relevant testing

ADDS - 

POST /api/articles/:article_id/comments endpoint 
relevant testing for endpoint ^ 

UPDATES - 

endpoints.json with the new endpoint

### CODE REVIEW

File: models/article-models/article-models.js [hunk 1]
```diff
@@ -88,3 +88,59 @@ exports.getComments = (id) => {
...
+    Object.keys(body).length === 2 &&
+    typeof username === "string" &&
+    typeof commentBody === "string"
+  ) {
+    const articleID = parseInt(id);
+    if (isNaN(articleID) || articleID <= 0 || String(articleID) !== id) {
+      return Promise.reject({
+        status: 400,
+        msg: "Invalid endpoint / article ID",
+      });
```
Comment (nc-john-hutcheon) [inline models/article-models/article-models.js:108]: It's really great that you're considering all of this validation, but SQL will actually do a lot of it for us and throw an error itself, so there's no requirement for a lot of this code! 😄

**Review by nc-john-hutcheon — COMMENTED** [review]
Hi Matt, you've got some good work here but have definitely put too much code in your model!
SQL will throw a lot of errors for us here, e.g...

- author in the comments table is a foreign key which relates to username in the users table. A psql error will be thrown if this is violated. It is not necessary to separately check if the user exists.
- article_id is also a foreign key so again, a separate check that the article exists should not be necessary as a psql error will be thrown.
- a psql error will also be thrown if article_id is not a number.

I'd refactor your model in line with this so there's less code in there. Your tests are all great, but may be slightly overkill and considering edge cases. As a general rule, the tests should just be...

- Status 201, created comment object
- Status 201, ignores unnecessary properties
- Status 400, invalid ID, e.g. string of "not-an-id"
- Status 404, non existent ID, e.g. 0 or 9999
- Status 400, missing required field(s), e.g. no username or body properties
- Status 404, username does not exist

You've got all of these (just need to refactor one of your 400s to be a 201), so just try and clean this up a bit and you should be fine 🙂
